components/plan-build/bin/hab-plan-build.sh

```diff
@@ -1603,6 +1603,25 @@ _resolve_dependencies() {
   _validate_deps
 }
 
+function do_automatic() {
```

Comment:
I'd argue that we want to make this an internal/private function only and not a user-facing part of our build API. Naming this `_do_automatic` or even possibly `_load_automatic` would be okay. In that case, we wouldn't need a default implementation of `do_default_automatic`--that's just how the build program works (just like we don't have a default implementation for setting the PATH because users shouldn't override this).